app/controllers/course/assessment/submission/submissions_controller.rb

```diff
@@ -298,14 +298,14 @@ def download_job
   # Check for zombie jobs, create new grading jobs if there's any zombie jobs.
   # TODO: Remove this method after found the cause of the dead jobs.
   def check_zombie_jobs # rubocop:disable Metrics/AbcSize, Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
-    return unless @submission.attempting?
+    return unless @submission.attempting? || @submission.submitted?
 
-    submitted_answers = @submission.answers.latest_answers.select(&:submitted?)
+    submitted_answers = @submission.answers.select(&:submitted?)
     return if submitted_answers.empty?
 
     dead_answers = submitted_answers.select do |a|
       job = a.auto_grading&.job
-      job&.submitted? && job.created_at < Time.zone.now - 60.minutes
+      job&.submitted? && !job.in_queue?
```

Comment:
Seems like this is the definitive definition of a dead job: a job that's submitted but not in any queue. But `in_queue?` is not exactly dirt cheap compared to the previous 1-hour check since it has to do a nested search across all queues for `job`. That's probably the trade-off; am I on the right track?

So, is it not the reason why you added that 5 minutes buffer not for performance, but instead to account for the possibility of a job taking a while to be enqueued? Or jobs are usually enqueued immediately by Sidekiq with no/negligible delays?